ui/components/app/modals/export-private-key-modal/export-private-key-modal.component.js

```diff
@@ -14,101 +21,121 @@ import {
+  hideWarning,
+  exportAccount,
+  selectedIdentity,
+  showAccountDetailModal,
+  hideModal,
+  warning = null,
+  previousModalState,
+}) => {
+  const [password, setPassword] = useState('');
+  const [privateKey, setPrivateKey] = useState(null);
+  const [showWarning, setShowWarning] = useState(true);
+  const [showHoldToReveal, setShowHoldToReveal] = useState(false);
 
-  componentWillUnmount() {
-    this.props.clearAccountDetails();
-    this.props.hideWarning();
-  }
+  const trackEvent = useContext(MetaMetricsContext);
+  const t = useI18nContext();
 
-  exportAccountAndGetPrivateKey = (password, address) => {
-    const { exportAccount } = this.props;
+  useEffect(() => {
+    return () => {
+      clearAccountDetails();
+      hideWarning();
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
 
-    exportAccount(password, address)
-      .then((privateKey) => {
-        this.context.trackEvent({
+  const exportAccountAndGetPrivateKey = async (passwordInput, address) => {
+    try {
+      const privateKeyRetrieved = await exportAccount(passwordInput, address);
+      trackEvent(
+        {
           category: MetaMetricsEventCategory.Keys,
           event: MetaMetricsEventName.KeyExportRevealed,
           properties: {
             key_type: MetaMetricsEventKeyType.Pkey,
           },
-        });
-
-        this.setState({
-          privateKey,
-          showWarning: false,
-        });
-      })
-      .catch((e) => {
-        this.context.trackEvent({
+        },
+        {},
+      );
+      setPrivateKey(privateKeyRetrieved);
+      setShowWarning(false);
+      setShowHoldToReveal(true);
+    } catch (e) {
+      trackEvent(
+        {
           category: MetaMetricsEventCategory.Keys,
           event: MetaMetricsEventName.KeyExportFailed,
           properties: {
             key_type: MetaMetricsEventKeyType.Pkey,
             reason: 'incorrect_password',
           },
-        });
+        },
+        {},
+      );
 
-        log.error(e);
-      });
+      log.error(e);
+    }
   };
 
-  renderPasswordLabel(privateKey) {
+  const renderPasswordLabel = (privateKeyInput) => {
     return (
-      <span className="export-private-key-modal__password-label">
-        {privateKey
-          ? this.context.t('copyPrivateKey')
-          : this.context.t('typePassword')}
-      </span>
+      <Text
+        as="span"
+        color={Color.textDefault}
+        marginBottom={2}
+        variant={TextVariant.bodySm}
+      >
+        {privateKeyInput ? t('copyPrivateKey') : t('typePassword')}
+      </Text>
     );
-  }
+  };
 
-  renderPasswordInput(privateKey) {
-    const plainKey = privateKey && stripHexPrefix(privateKey);
+  const renderPasswordInput = (privateKeyInput) => {
+    const plainKey = privateKeyInput && stripHexPrefix(privateKeyInput);
 
-    if (!privateKey) {
+    if (!privateKeyInput) {
       return (
         <input
+          aria-label="input-password"
```

Comment:
What is the aria-label used for here? If we used a placeholder it would be more appropriate? 
```suggestion
          placeholder={t('enterPassword')} // not actually sure this exists in locales
```

Reply:
sure, let me change that. the placeholder value does exist in locales